app/views/batch_contexts/_new_bc_form.erb

```diff
@@ -24,20 +24,37 @@
         %>
     <% end %>
 
-    <%= form.input :content_structure, collection:
-    [
-        ['Image', 'simple_image'],
-        ['Book (ltr)', 'simple_book'],
-        ['Book (rtl)', 'simple_book_rtl'],
-        ['Document', 'document'],
-        ['File', 'file'],
-        ['Geo', 'geo'],
-        ['Media', 'media'],
-        ['3D', '3d'],
-        ['Map', 'maps'],
-        ['Webarchive seed', 'webarchive_seed']
-    ]
-    %>
+    <% if Settings.ocr.enabled # if OCR is enabled, the content_structure drop down menu will hide/show ocr menus via stimulus %>
+        <%= form.input :content_structure, input_html: { data: { action: "change->ocr#changed", "ocr-target": "contentStructure" } }, collection: content_structure %>
+
+        <div id="ocr-settings-image" data-ocr-target="ocrSettingsImage" hidden>
+            <%= form.input :manually_corrected_ocr,
+                as: :radio_buttons,
+                input_html: { data: { "ocr-target": "manuallyCorrectedOcrImage" } },
+                label: 'Do the image(s) have pre-existing OCR that has been manually reviewed/corrected for accessibility?',
+                collection: [['Yes', true], ['No', false]] %>
+            <%= form.input :run_ocr,
+                as: :radio_buttons,
+                input_html: { data: { "ocr-target": "runOcrImage" } },
+                label: 'Would you like to run SDR/ABBYY OCR on your image(s)?',
+                collection: [['Yes. Run ABBYY OCR.', true], ['No. Do not run ABBYY OCR.', false]] %>
+        </div>
+
+        <div id="ocr-settings-document" data-ocr-target="ocrSettingsDocument" hidden>
+            <%= form.input :manually_corrected_ocr,
+                as: :radio_buttons,
+                input_html: { data: { "ocr-target": "manuallyCorrectedOcrDocument" } },
+                label: 'Have you manually reviewed/corrected your document(s) for WCAG/PDFUA compliance?',
+                collection: [['Yes. Document(s) meet a11y requirements. You don’t need to run SDR/ABBYY OCR.', true], ["No/Don't Know. We recommend that you manually review/correct the document in an editor, and then return to Preassembly.", false]] %>
+            <%= form.input :run_ocr,
+                as: :radio_buttons,
+                input_html: { data: { "ocr-target": "runOcrDocument" } },
+                label: 'Would you like to run SDR/ABBYY OCR on your image(s)?',
+                collection: [['Yes. Run ABBYY OCR. Warning: If your document is already WCAG/PDFUA compliant,  ABBYY may create a less accessible version.', true], ['No. Do not run ABBYY OCR.', false]] %>
+        </div>
```

Comment:
so here is an issue, the radio buttons are selectively hidden/shown based on the content structure selected (image vs document) so only one is visible and enabled at the same time

but they have the same names ("run_ocr" and "manually_corrected_ocr") since they can be stored the same in the model

however, even though only one matching set is enabled at a time (in the HTML dom), the rails params is not correctly picking up the values for the first set (if you select "Image") ... it just gives nil

not sure if there is a clever solution to this

Reply:
maybe i need one copy of the radio buttons and selectively change the labels in stimulus?